### PR TITLE
added trim() to string parser

### PR DESCRIPTION
**Pull request by yuvalherziger**
Usage:

```php
// full trim:
$this->queryParameter('someQueryParam')->string()->trim()->required();
// right trim:
$this->queryParameter('anotherQueryParam')->string()->rightTrim()->required();
// left trim:
$this->bodyParameter('someBodyParam')->string()->leftTrim()->required();
```

### CODE REVIEW

File: src/TrimParser.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    public function __construct(Config $config, $name, $value, $trimType)
+    {
+        $this->trimType = $trimType;
+        parent::__construct($config, $name, $value);
+    }
+
+    protected function describe()
+    {
+        return "a valid email address";
```
Comment (mpscholten) [inline src/TrimParser.php]: Looks like a copy paste mistake :) Should be like https://github.com/mpscholten/request-parser/blob/master/src/StringParser.php#L9

File: src/TrimType.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace MPScholten\RequestParser;
+
+class TrimType
+{
+    const TRIM = 'trim';
```
Comment (mpscholten) [inline src/TrimType.php]: Can we move these constants into the `TrimParser` as this class cannot exist without `TrimParser` (and the `TrimParser` can also not exist without the `TrimType`) ? :)

File: tests/TypeSpecTest.php [hunk 3]
```diff
@@ -57,6 +58,12 @@ public function testEmail()
         $this->assertInstanceOf(EmailParser::class, $spec->string()->email());
     }
 
+    public function testTrim()
+    {
+        $spec = new TypeParser(new Config(), 'emailAddress', '   john@doe.com   ');
+        $this->assertInstanceOf(TrimParser::class, $spec->string()->trim());
```
Comment (mpscholten) [inline tests/TypeSpecTest.php:64]: Can we also have a small test for the left and right trim? :) Just to have everything covered

**Comment by yuvalherziger** [comment]
@mpscholten would be cool if you could review this when time permits :-)

**Comment by mpscholten** [comment]
Just saw that I forgot to submit the review :D it was already here, but still in the pending state

File: src/TrimType.php (hunk 2 above)
Comment (yuvalherziger) [reply src/TrimType.php]: Good point 👍

File: tests/TypeSpecTest.php (hunk 3 above)
Comment (yuvalherziger) [reply tests/TypeSpecTest.php:64]: Makes sense :-)

**Comment by mpscholten** [comment]
Looks good to me 👍  good job

**Pull request merged**